code/datums/elements/wall_smasher.dm

```diff
@@ -0,0 +1,61 @@
+/**
+ * # Wall Smasher
+ * An element you put on mobs to let their attacks break walls
+ */
+/datum/element/wall_smasher
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	/// Whether you can smash RWalls or not
+	var/strength_flag
+
+/datum/element/wall_smasher/Attach(datum/target, strength_flag = ENVIRONMENT_SMASH_WALLS)
+	. = ..()
+
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.strength_flag = strength_flag
+	RegisterSignal(target, COMSIG_LIVING_UNARMED_ATTACK, PROC_REF(on_unarm_attack)) // Players
+	RegisterSignal(target, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(on_pre_attackingtarget)) // AI
+
+/datum/element/wall_smasher/Detach(datum/target)
+	UnregisterSignal(target, list(COMSIG_LIVING_UNARMED_ATTACK, COMSIG_HOSTILE_PRE_ATTACKINGTARGET))
+	return ..()
+
+/datum/element/wall_smasher/proc/on_unarm_attack(mob/living/puncher, atom/target, proximity, modifiers)
+	SIGNAL_HANDLER
+	try_smashing(puncher, target)
+
+/datum/element/wall_smasher/proc/on_pre_attackingtarget(mob/living/puncher, atom/target)
+	SIGNAL_HANDLER
+	try_smashing(puncher, target)
+
+/datum/element/wall_smasher/proc/try_smashing(mob/living/puncher, atom/target)
+	if (!isturf(target))
+		return
+	if (isfloorturf(target))
+		return
+	if (isindestructiblewall(target))
+		return
+
+	. = COMPONENT_HOSTILE_NO_ATTACK
```

Comment:
Everywhere after this you should `return .` so that it's obvious we are returning a value, or `return COMPONENT_HOSTILE_NO_ATTACK` explicitly.